frontend/src/features/contacts/contact/create/Person/CreatePersonForm.tsx

```diff
@@ -0,0 +1,262 @@
+import { AsyncTypeahead, Button, Input } from 'components/common/form';
+import { FormSection } from 'components/common/form/styles';
+import { UnsavedChangesPrompt } from 'components/common/form/UnsavedChangesPrompt';
+import GenericModal from 'components/common/GenericModal';
+import { Stack } from 'components/common/Stack/Stack';
+import { CountryCodes } from 'constants/countryCodes';
+import { Address, ContactEmailList, ContactPhoneList } from 'features/contacts/contact/create';
+import { personCreateFormToApiPerson } from 'features/contacts/contactUtils';
+import {
+  Formik,
+  FormikHelpers,
+  FormikProps,
+  getIn,
+  validateYupSchema,
+  yupToFormErrors,
+} from 'formik';
+import { useApiAutocomplete } from 'hooks/pims-api/useApiAutocomplete';
+import { useApiContacts } from 'hooks/pims-api/useApiContacts';
+import { IAutocompletePrediction } from 'interfaces';
+import { defaultCreatePerson, ICreatePersonForm } from 'interfaces/ICreateContact';
+import { useMemo, useState } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { AiOutlineExclamationCircle } from 'react-icons/ai';
+import { useHistory } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+import useAddressHelpers from '../address/useAddressHelpers';
+import { PadBox } from '../styles';
+import CommentNotes from './comments/CommentNotes';
+import * as Styled from './styles';
+import { hasAddress, hasEmail, hasPhoneNumber, validationSchema } from './validation';
+
+export interface ICreatePersonFormProps {}
+
+/**
+ * Formik-connected form to Create Individual Contacts
+ */
+export const CreatePersonForm: React.FunctionComponent<ICreatePersonFormProps> = props => {
+  const history = useHistory();
+  const { postPerson } = useApiContacts();
+
+  // validation needs to be adjusted when country == OTHER
+  const { countries } = useAddressHelpers();
+  const otherCountryId = useMemo(
+    () => countries.find(c => c.code === CountryCodes.Other)?.value?.toString(),
+    [countries],
+  );
+
+  const onValidate = (values: ICreatePersonForm) => {
+    try {
+      validateYupSchema(values, validationSchema, true, { otherCountry: otherCountryId });
+      // combine yup schema validation with custom rules
+      const errors = {} as any;
+      if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
+        errors.needsContactMethod =
+          'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
+      }
+      return errors;
+    } catch (err) {
+      return yupToFormErrors(err);
+    }
+  };
+
+  const onSubmit = async (
+    values: ICreatePersonForm,
+    { resetForm, setSubmitting }: FormikHelpers<ICreatePersonForm>,
+  ) => {
+    try {
+      await postPerson(personCreateFormToApiPerson(values));
+      setSubmitting(false);
+      resetForm({ values });
```

Comment:
does this make sense? the rowversion will be off on the form if you do this. Also if the api enforces any logic on the request the FE won't see that.

Reply:
this is the create form not the update form so rowversion would always be undefined no? In essence I had to reset the form because the next line is a route change and if I didn't reset the form I would get the router guard saying that I had unsaved changes. Would like to hear how you solved this as I used the same router guard code you are using

Reply:
I have the prompt in context of formik, and so if the form is submitted successfully I have a flag that prevents the route guard from triggering.

Reply:
okay, I can change in the next PR to check for that. What flag you use?

Reply:
Found the flag and made the change: `submitCount === 0`